web/src/components/molecules/Common/Form/AssetItem/hooks.ts

```diff
@@ -24,25 +28,42 @@ export default (
     setUploadModalVisibility(true);
   }, [setUploadModalVisibility]);
 
-  const hideUploadModal = useCallback(() => {
-    setUploadModalVisibility(false);
-    setUploading(false);
-    setFileList([]);
-  }, [setFileList, setUploadModalVisibility, setUploading]);
-
   const handleUpload = useCallback(async () => {
     setUploading(true);
-    const assets = await createAssets(fileList);
-    if (assets?.length > 0) onLink(assets[0]);
-    hideUploadModal();
-  }, [createAssets, fileList, hideUploadModal, onLink, setUploading]);
+    let assets: (Asset | undefined)[] = [];
+    let asset: Asset | undefined;
```

Comment:
I can see what you are doing, but this callback is `handleUpload`. It shouldn't also have logic around linking. Using `asset` as the way to know whether to link or not isn't obvious. 

Also, is the only reason you are saving `uploadUrl` to state to be used in this callback? If so, please just pass the values to the callback directly. If you have issues around getting the value from antd's form or soemthing, and that is the main reason you decided to use state, just make the state locally in the form's component and pass the values of that state to this callback as props.

Reply:
Please let me know what do you think.